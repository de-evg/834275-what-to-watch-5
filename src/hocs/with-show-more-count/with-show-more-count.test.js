import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withShowMoreCount from "./with-show-more-count";
const SHOW_MORE_STEP = 8;

const noop = () => {};

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withShowMoreCount(MockComponent);

it(`withShowMoreCount is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      onShowMoreClick={noop}
      resetShowedMovies={noop}
      showedMoviesCount={SHOW_MORE_STEP}
    >
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
