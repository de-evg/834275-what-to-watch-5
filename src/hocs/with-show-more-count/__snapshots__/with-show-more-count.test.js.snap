// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withShowMoreCount is rendered correctly 1`] = `<div />`;
